spec/models/deal_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+RSpec.describe Deal, type: :model do
+  subject { described_class.new(title: 'abcd', description: 'This is a random deal', price_in_cents: 10000, discount_price_in_cents: 9000, quantity: 12, tax_percentage: 10, publish_at: Time.current) }
```

Comment:
read about how `subject` works in different scenarios when `string|class|number` is passed to `RSpec.describe`